hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/Planner.java

```diff
@@ -67,15 +67,15 @@ DAG createDag() {
         // Find the greatest common denominator of all frame lengths
         // appearing in the pipeline
         long frameSizeGcd = Util.gcd(adjacencyMap.keySet().stream()
-                                                 .map(Transform::watermarkFrameSize)
+                                                 .map(Transform::preferredWatermarkStride)
                                                  .filter(frameSize -> frameSize > 0)
                                                  .mapToLong(i -> i)
                                                  .toArray());
         // Update watermark emission policy on all wm gen params
         // with the GCD frame length
         WatermarkEmissionPolicy emitPolicy = frameSizeGcd > 0
                 ? emitByFrame(tumblingWinPolicy(frameSizeGcd))
-                : noThrottling();
+                : noWatermarks();
```

Comment:
we already have WatermarkGenerationParams.noWatermarks() , this seems like a duplication?

Reply:
This code was simpler with it, but now I'm replacing the entire `wmParams` instead.